navbar: Scale navbar with app font size, using em.

evykassirer opened the pull request:
<details>
<summary>screenshots: simple navbar</summary>

12px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/a38e175f-bcca-4fe3-bbdf-f8736c2b92fd" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/17902c45-07ac-487c-a300-55edf023f7dd" />


14px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/777e6e9e-05e9-4748-a653-ce088c584df3" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/bff293ec-9f9e-4835-a8ea-342ee28a0eca" />


16px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/9be61447-d4e6-4190-aaa3-31fc5a5bcbb0" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/a146a9b9-0c97-48b8-a56a-3c8baad993aa" />


20px before after

<img width="616" alt="image" src="https://github.com/user-attachments/assets/d09e2040-6848-4fd1-b673-3f1413841ea2" />

<img width="616" alt="image" src="https://github.com/user-attachments/assets/4d9a6c41-2ff1-46fc-b50b-762f173de32e" />

</details>





<details>
<summary>screenshots: search with pills</summary>

12px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/55c12bf4-9fb5-488d-be7f-2486e5bebe72" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/c6f9d6a3-7ae7-47e5-a41c-a6c3bc36fb01" />

14px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/c90c4840-44a8-4d62-88c7-2ab206c532f1" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/3277cb5b-f6bd-45e9-a391-30ea470477b8" />

16px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/8e5e2f33-17ae-45f7-b8c0-412017cb0c70" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/bb5a12cd-01b3-4f1c-8062-60ff2b12c8dc" />

20px before and after

<img width="618" alt="image" src="https://github.com/user-attachments/assets/7d29b1dc-667a-48b4-9810-dbdac4bc470c" />

<img width="619" alt="image" src="https://github.com/user-attachments/assets/7cecb93c-5df7-4864-ac7d-ca9a53d382e2" />


</details>





<details>
<summary>screenshots: short navbar</summary>

12px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/f9f35ea4-151f-406a-980a-fbcb56960abf" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/f9027a0a-8189-4786-8245-281d9d83a67a" />


14px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/a447cd1d-a1d7-497f-a976-9690efd75685" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/66b55ade-fc8d-432f-9edc-26a6789fe49e" />


16px before and after

<img width="619" alt="image" src="https://github.com/user-attachments/assets/67dba76c-44da-4058-8eb1-6f253851478a" />

<img width="619" alt="image" src="https://github.com/user-attachments/assets/629f86eb-f824-41ef-b521-49037cae0966" />


20px before and after

<img width="617" alt="image" src="https://github.com/user-attachments/assets/b54deb6b-cf27-483c-9542-fc808c3010fb" />

<img width="617" alt="image" src="https://github.com/user-attachments/assets/22b3c897-0547-4606-add6-d25c16eb47fc" />



</details>

karlstolley commented on web/styles/search.css:
```diff
@@ -13,8 +13,8 @@
         text-decoration: none;
-        padding: 0 7px;
-        font-size: 17px;
+        padding: 0 0.5em; /* 7px at 14px em */
```
This calculation is wrong. Because the font-size is being set to the em equivalent of 17px, that's the em value needed here: `/* 7px at 17px/em */

karlstolley commented:
Although our precedent of using 14px as the em value to do these kinds of conversions on legacy parts of the UI is well established at this point, I wonder if the navbar items here would be better set to 16px for the em. It seems pretty well right-sized to me at 16px at the moment, and this would have the effect of blowing it up larger in ways that I'm not sure are ideal.

I've also added a comment as to why the message headers are shrinking even as the navbar gets larger.

evykassirer commented:
> I've also added a comment as to why the message headers are shrinking even as the navbar gets larger.

I only see a comment on search icon padding, and updated that. Is there something to change around the message headers here too?

> I wonder if the navbar items here would be better set to 16px for the em. It seems pretty well right-sized to me at 16px at the moment, and this would have the effect of blowing it up larger in ways that I'm not sure are ideal.

I could go through and scale a bunch of them by 16px instead if there's desire for that. Maybe first @alya can weigh in on if she's interested in seeing that before we merge this, or if it would be better prioritized as a possible followup?

alya commented:
I don't understand the question, so might be best to ask in #design with more explanation.

karlstolley commented on web/styles/message_header.css:
```diff
@@ -6,13 +6,13 @@
     text-overflow: ellipsis;
     white-space: nowrap;
     font-weight: 600;
-    font-size: calc(15em / 14);
```
I'm kind of confused why this commit is in with the navbar stuff. Seems like it should be its own PR, and probably ought to address other icons and elements that are children of `.message_header`

timabbott replied:
I'd definitely prefer a separate PR with screenshots. Message headers are a very high importance surface area, and we want to be very careful to avoid breaking anything involving them.

evykassirer replied:
33185

evykassirer replied:
Merged! I just pushed this PR with out this commit.

karlstolley commented on web/styles/message_header.css:
```diff
@@ -6,13 +6,13 @@
     background-color: var(--color-background);
 
     .message-header-contents {
+        font-size: 1.0714em; /* 15px at 14px em */
```
This font-size change probably changes the em-value on the `height` property below.

karlstolley replied:
In fact this may be why the before and after shots at e.g., 16px look off to me. This also throw into question whether 14px is the right em value to target for these adjustments, or 16px--similar to the nav bar.

evykassirer replied:
Wait is this a font-size change? I was under the impression that `calc(15em / 14)` is functionally exactly the same as `1.0714em`

evykassirer replied:
But the height should be based on 15px, right? I believe that's the followup needed from https://github.com/zulip/zulip/pull/32933/files#r1910137015

karlstolley replied:
It's functionally the same, yes, but any time `font-size` is set to anything other than `1em`, that changes the em value for other properties in the style block and its children.

evykassirer replied:
Right, but since the parent already had this font-size in em then it was still true that this style block and its children had the different em value.

karlstolley replied:
I'm talking about the `height: 2em; /* 28px at 14px em */`...perhaps what's needed there is just the 15px calculation, meaning the comment should be `/* 30px at 15px/1em */`

evykassirer replied:
Ah yeah, definitely the height issue is messed up, but that happened already in a previous PR and this PR doesn't exacerbate that issue. I see this PR as a code differential with no visible differences, and #33192 as fixing a bug introduced in #32933. So I think it makes sense to continue that discussion there, unless I'm still missing something.

evykassirer commented:
I'm pushing a commit with the change to 16px default height. Here are some before (14px default) / after (16px default) screenshots

at 12px:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/dad6fc07-5215-4269-841b-958fa53a1b6a" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/1ee40934-72d6-4482-ba99-51766a57fc21" />

at 14px:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/497296f7-7bc2-485d-a7b7-6e8395d08fac" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/e01ce63d-fa5e-449f-92ee-9cd914470cca" />

at 16px:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/a43992cc-1984-46db-b0b2-f76ac3a497b7" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/f4dc702a-6437-4e24-8f12-611a674f72eb" />


at 20px:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/68165708-5ab1-408e-9b92-383539132d6b" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/d2f80979-0837-4f54-a3f7-f22e541322be" />

karlstolley commented:
Okay. [Those screenshots](https://github.com/zulip/zulip/pull/32928#issuecomment-2613810496) were not very good for comparing before/after, so I made a set of my own at 14px, 16px, and 20px:

| Before | After |
| --- | --- |
| ![navbar-14-before](https://github.com/user-attachments/assets/8c49ceca-f3a7-40c9-b5ac-ab7c3ac13f9f) | ![navbar-14-after](https://github.com/user-attachments/assets/85fd847a-4aa3-4ebe-a62c-805ae8044604) |
| ![navbar-16-before](https://github.com/user-attachments/assets/675da93b-912f-4a1b-b1e7-716ead116ec4) | ![navbar-16-after](https://github.com/user-attachments/assets/ba7d01f7-f7ea-4d73-a95e-1303d14bb5fd) |
| ![navbar-20-before](https://github.com/user-attachments/assets/a895b461-f049-4f43-9b23-135377764fc5) | ![navbar-20-after](https://github.com/user-attachments/assets/b392df9d-40ac-4a8b-9c77-139c007e1580) |

There are some subtle alignment shifts with the Search box at 14px and 20px, and this hopefully gives a better look at the impact of the icon resizing adjustments here.

evykassirer commented:
Okay updated the last commit to change _everything_ to 16px, instead of previously just the thing I felt were more notable 😅 

If we like 16px better for sure, I'll squash that into the previous commit (unless anyone would prefer I don't?)

Here's 14px, 16px, 20px. You can see 16px now has no change.

| before | after |
| ------------- | ------------- |
| ![Screen Shot 2025-01-28 at 16 02 19](https://github.com/user-attachments/assets/932bb5e3-c198-4029-93f6-a2eeb90c19e1) | ![Screen Shot 2025-01-28 at 16 01 15](https://github.com/user-attachments/assets/225f9a8a-66d2-4c93-be71-43f87129ba86) |
| ![Screen Shot 2025-01-28 at 16 02 32](https://github.com/user-attachments/assets/6c359bc7-27a5-4e16-b7fb-c309a5b73b7a) | ![Screen Shot 2025-01-28 at 16 01 05](https://github.com/user-attachments/assets/df35b607-1930-44f9-af81-7056d1051dcc) |
| ![Screen Shot 2025-01-28 at 16 02 41](https://github.com/user-attachments/assets/b8bfca75-ff24-40db-b497-d36bdeca4002) | ![Screen Shot 2025-01-28 at 16 00 52](https://github.com/user-attachments/assets/944d4250-3cd9-4300-8df7-f0ad0a324ccc) |

alya commented:
Seems plausible to me based on the screenshots (but needs @karlstolley 's review).

karlstolley commented:
The 16px looks good, but the interstitial commit using 14px as the base em should be pulled/squashed in favor of the 16px conversion. Once that's done, I think this should be ready to merge.

timabbott commented:
Looks good, marked to merge once CI passes. Thanks @evykassirer and @karlstolley!
